main.tf

```diff
@@ -0,0 +1,692 @@
+resource "azurerm_network_security_group" "nsg_businessTier" {
+  name                = "businessTierNetworkSecurityGroup"
+  location            = "East US"
+  resource_group_name = azurerm_resource_group.resource_group.name
+}
+resource "azurerm_network_security_rule" "nsr_businessTier1" {
+  name                        = "RDP3389"
+  priority                    = 1001
+  direction                   = "Inbound"
+  access                      = "Allow"
+  protocol                    = "Tcp"
+  source_port_range           = "*"
+  destination_port_range      = "3389"
+  source_address_prefix       = "10.0.1.0/24"
+  destination_address_prefix  = "10.0.2.0/24"
+  resource_group_name         = azurerm_resource_group.resource_group.name
+  network_security_group_name = azurerm_network_security_group.nsg_businessTier.name
+}
+resource "azurerm_network_security_rule" "nsr_businessTier2" {
+  name                        = "RDP_webTier_3389"
+  priority                    = 1002
+  direction                   = "Inbound"
+  access                      = "Allow"
+  protocol                    = "Tcp"
+  source_port_range           = "*"
+  destination_port_range      = "3389"
+  source_address_prefix       = "10.0.0.0/27"
+  destination_address_prefix  = "10.0.2.0/24"
+  resource_group_name         = azurerm_resource_group.resource_group.name
+  network_security_group_name = azurerm_network_security_group.nsg_businessTier.name
+}
+resource "azurerm_subnet_network_security_group_association" "businesTierSubnetAssociation" {
+  subnet_id                 = azurerm_subnet.businessTierSubnet.id
+  network_security_group_id = azurerm_network_security_group.nsg_businessTier.id
+}
+# Create NSG and Network Security rules for Data Tier.
+resource "azurerm_network_security_group" "nsg_dataTier" {
+  name                = "dataTierNetworkSecurityGroup"
+  location            = "East US"
+  resource_group_name = azurerm_resource_group.resource_group.name
+}
+resource "azurerm_network_security_rule" "nsr_dataTier1" {
+  name                        = "RDP3389"
+  priority                    = 1001
+  direction                   = "Inbound"
+  access                      = "Allow"
+  protocol                    = "Tcp"
+  source_port_range           = "*"
+  destination_port_range      = "3389"
+  source_address_prefix       = "10.0.2.0/24"
+  destination_address_prefix  = "10.0.3.0/24"
+  resource_group_name         = azurerm_resource_group.resource_group.name
+  network_security_group_name = azurerm_network_security_group.nsg_dataTier.name
+}
+resource "azurerm_network_security_rule" "nsr_dataTier2" {
+  name                        = "RDP_businessTier_3389"
+  priority                    = 1002
+  direction                   = "Inbound"
+  access                      = "Allow"
+  protocol                    = "Tcp"
+  source_port_range           = "*"
+  destination_port_range      = "3389"
+  source_address_prefix       = "10.0.0.0/27"

```

Comment:
These addresses duplicates with the subnet definitions. You can use local variables to define these ranges in one place.

https://www.terraform.io/docs/configuration/locals.html